### DISCUSSION TITLE
DB2 DDL does not work depending on db2 jdbc version

### DISCUSSION THREAD

**Discussion by Axidi** [dbeaver/dbeaver#11157] [Q&A]
Hello,
Happy new year !

I have discovered that getting the DDL for tables on my db2 database does not work anymore.
I used different DB2 users with not success.
So I tried older db2 jdbc driver version:

**It does work with driver 4.25.1301 but starting with 4.26.14 it does not anymore**

I use DB2 database version 11.5

Your tool is great , please continue this way

Regards
Alex

#### System information: 
- Operating system (distribution) and version
- DBeaver version
- Additional extensions

#### Connection specification:
- Database name and version
- Driver name
- Do you use tunnels or proxies (SSH, SOCKS, etc)?

**Comment by kseniiaguzeeva** [comment]
Thank you for your report. Unfortunately I do not have an environment to reproduce it. What do you mean when tell "does not work":
- Doesn't it appear when you press DDL tab in table properties or when you use Generate DDL option?
- Or does it have a wrong syntax?
- Do any errors appear?

Please tell me also DBeaver version and send log files https://github.com/dbeaver/dbeaver/wiki/Log-files

**Comment by Axidi** [comment]
Hello,

Thanks for your repply.
I use Dbeaver version 7.3.2.

When I select a table and then "Generate SDL" / DDL, I receive the message :
_Routine "*RATE_DDL" (specific name "") has returned an error SQLSTATE with diagnostic text "SQL0551 Reason code or token: DB2SFNP,CREATE TABLE,SYSTOOLS".. SQLCODE=-443, SQLSTATE=38553, DRIVER=4.27.25_

I have attached the log file.

Regards
Alex

[.log](https://github.com/dbeaver/dbeaver/files/5795284/default.log)

**Comment by kseniiaguzeeva** [comment]
Please tell which driver did you select to connect? 
![Screenshot 2021-01-19 at 15 51 13](https://user-images.githubusercontent.com/51405061/105037171-a53be380-5a6e-11eb-8754-e03aca362268.png)
Did you make any driver changes or other additional settings?
![Screenshot 2021-01-19 at 15 54 32](https://user-images.githubusercontent.com/51405061/105037188-aa009780-5a6e-11eb-86c9-63a33379944e.png)

**Comment by Axidi** [comment]
**The db2 jdbc driver which works here a capture**
![image](https://user-images.githubusercontent.com/50483555/105888946-e5223e00-600d-11eb-8b55-9aa85c517fc8.png)

**The db2 jdbc driver which does not work here a screen capture**
![image](https://user-images.githubusercontent.com/50483555/105889195-33cfd800-600e-11eb-8507-cd25f033d676.png)

Tell please if you need more informations.

**Comment by Axidi** [comment]
No I did not make change or additional settings on the driver.

**Comment by kseniiaguzeeva** [comment]
As I see DB2 z/OS works for you database, and DB2 LUW doesn't work. One additional detail is needed: please press Test Connection and send me the capture with the result in popup, like this:
![dbeaver_uD8QnZATHr](https://user-images.githubusercontent.com/51405061/106105990-ec546380-6155-11eb-8462-c1977dd550ea.png)

**Comment by Axidi** [comment]
I created a new connection with DB2 z/OS and it works!
Here the captures

DDL OK
![image](https://user-images.githubusercontent.com/50483555/106141970-2848f200-6171-11eb-9a9d-476523599378.png)

DDL not OK
![image](https://user-images.githubusercontent.com/50483555/106142152-69d99d00-6171-11eb-9739-622bdfdef3e8.png)

Thanks for your help.

dbeaver ist a great tool!

Regards
Alex

**Comment by kseniiaguzeeva** [comment]
Nice to hear! Can I close the ticket?

**Comment by Axidi** [comment]
Yes please,
Thanks for your help!
